boot/bootutil/src/loader.c

```diff
@@ -1987,9 +2168,22 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
     uint32_t i;
     int fa_id;
     int rc;
+#ifdef MCUBOOT_RAM_LOADING
+    uint32_t img_dst;
+    uint32_t img_sz;
+    uint32_t img_loaded = 0;
+#endif /* MCUBOOT_RAM_LOADING */
 
     memset(state, 0, sizeof(struct boot_loader_state));
 
+#ifdef MCUBOOT_RAM_LOADING
+    static boot_sector_t primary_slot_sectors[BOOT_MAX_IMG_SECTORS];
+    static boot_sector_t secondary_slot_sectors[BOOT_MAX_IMG_SECTORS];
+
+    BOOT_IMG(state, BOOT_PRIMARY_SLOT).sectors = primary_slot_sectors;
+    BOOT_IMG(state, BOOT_SECONDARY_SLOT).sectors = secondary_slot_sectors;
+#endif /* MCUBOOT_RAM_LOADING */
```

Comment:
As I see it only required to get the sector size in the boot_copy_image_to_sram (assuming a homogeneous layout).
Would it be enough to calculate a sector size from the BOOT_MAX_IMG_SECTORS macro and the size of the image?
It would allow to remove all these extra steps (and functions).

Reply:
That could be an optimisation, but the support of inhomogeneous flash layout would be lost. I think RAM_LOADING will be enabled in a bit more powerful devices where the usage of extra RAM and CPU cycle is not issue. I would keep in this way to be consistent with other upgrade modes.

Reply:
As we discussed, the boot_copy_image_to_sram() function only reads from the flash, therefore it doesn't necessarily have to know the sector layout. It could use the 'BOOT_TMPBUF_SZ' constant similarly to the bootutil_img_hash() function.
All this would simplify the code and reduce RAM/flash usage somewhat.